cdap-common/src/main/java/co/cask/cdap/common/service/RetryOnStartFailureService.java

```diff
@@ -40,6 +39,7 @@
   private final Thread startupThread;
   private final String delegateServiceName;
   private volatile Service currentDelegate;
+  private volatile boolean stopped = false;
```

Comment:
Why need to be explicit? Java developer should familiar with Java 101, which includes non-final field default initialization.